src/rest-server/src/models/kubernetes.js

```diff
@@ -85,10 +97,50 @@ const getPods = async (options = {}) => {
   return res.data;
 };
 
+const createSecret = async (namespace, name, data, type) => {
+  const client = getClient();
+  const url = `/api/v1/namespaces/${namespace}/secrets`;
+  try {
+    const response = await client.post(url, {
+      metadata: {
+        name: name,
+        namespace: namespace,
+      },
+      data: data,
+      type: type,
+    });
+    if (response.status !== status('Created')) {
+      logger.warn('Failed to create secret');
+      throw createError(response.status, 'UnknownError', response.data.message);
+    }
+    logger.debug(`create secret ${name} successfully`);
+  } catch (err) {
+    logger.warn('Failed to create secret, err is: ', err);
+    throw err;
```

Comment:
pls check response body when axios response is not 20x, do not throw error directly here